elytron/WFCORE-4484-ssh-authentication-git-persistence.adoc

```diff
@@ -0,0 +1,140 @@
+= [WFCORE-4484] Support SSH authentication for Git persistence
+:author:            Ashley Abdel-Sayed
+:email:             aabdelsa@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+When using a Git repository to manage your WildFly configuration file history, it is possible to use an Elytron
+configuration file for authentication. However, it is currently only possible to specify credentials to authenticate
+with HTTP, not SSH, even though Git supports SSH authentication. This proposal is to add support for SSH authentication
+for Git persistence using an Elytron configuration file.
+
+It will be possible for users to specify location, filename (identity) and passphrase to an existing SSH private key to
+be used as credentials for SSH authentication.
+
+It will also be possible for users to specify the keys to be used for authentication as KeyPairCredentials. This will mean
+supporting specifying KeyPairCredentials in an Elytron configuration file as a credential or as a reference to an entry in
+a CredentialStore. This will require modifying the `key-pair` credential to accept the OpenSSH
+https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.key[private key] format. The user should then be able to
+configure their WildFly standalone server to connect to a Git repo and authenticate with SSH to manage their configuration file history.
+
```

Comment:
This feature might be interesting for Wildfly/EAP Openshift images where configuration might be taken from external git repository and not stored in image or created during container start.  

Such usage should be documented and quickstart could be added, wdyt?

Reply:
This sounds like a good idea, I'll add it to the documentation and look at working on a quickstart. Thank you!

Reply:
What about the quickstart? Will it be part of this RFE, or in a subsequent RFE?

Reply:
@darranl I was wondering if you think it would be better to include the quickstart as part of this RFE or hold off and create a follow up RFE? 

Also, as a side note when I was looking into testing out this feature with OpenShift, I was following these steps: https://github.com/wildfly/wildfly-s2i#building-the-images. I tried following the "Building WildFly s2i builder image with a locally built WildFly server" steps to test out my local changes for this RFE but I would get the following error:
```
 Non-resolvable import POM: Failure to find org.wildfly.core:wildfly-core-parent:pom:11.0.0.Beta6-SNAPSHOT in https://repository.jboss.org/nexus/content/groups/public/ was cached in the local repository, 
```
So I was wondering if you knew the steps to be able to use my local WildFly-Core changes to build the WildFly image?

Reply:
I think a follow up RFE could be created for the quickstart. @darranl What do you think?

Reply:
+1 any Quickstart should really be an RFE of it's own, however this RFE is only adding support for an authentication mechanism, this RFE does not add support for Git persistence so the Quickstart RFE should be against Management and not against Security.

Reply:
I've created https://issues.redhat.com/browse/EAP7-1521 to track the creation of a quickstart for git persistence.